subsys/net/ip/connection.c

```diff
@@ -657,8 +657,28 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 
 				mcast_pkt_delivered = true;
 			}
-		} else if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) ||
-			   IS_ENABLED(CONFIG_NET_SOCKETS_CAN)) {
```

Comment:
I'm not much familiar with CAN sockets, but do you think it makes more sense to omit them from this change, rather than include? I'd guess, they would have the same problem, e.g. if there're multiple CAN controllers (== interfaces), and a socket is bound to a particular interface, we want to receive packets just for that interface.

Reply:
There might be similar kind of issue for CAN sockets (would need some testing), but I do not want to start to change the CAN code here as I am not able to test the change.